packages/types/src/element/element.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import type { ElementBox } from './elementBox';
+
+export interface Link {
+  url: string;
+  desc?: string;
+  needsProxy?: boolean;
+  icon?: string;
+  rel?: string[];
+}
+
+export interface Flip {
+  vertical: number;
+  horizontal: number;
+}
+
+export interface Mask {
+  type: string;
+}
+
+export interface Border {
+  top: number;
+  right: number;
+  bottom: number;
+  left: number;
+  locked: boolean;
+}
+
+export interface BorderRadius {
+  topLeft: number;
+  topRight: number;
+  bottomLeft: number;
+  bottomRight: number;
+  locked: boolean;
+}
+
+export interface Element extends ElementBox {
```

Comment:
Maybe it would be good to add a union type of all the elements as well for using in other packages? We're very often using `element` as a prop without knowing which element type it is exactly. Maybe `element` could be a union type and `baseElement` (or sth similar) what it currently is?

Reply:
I see why you want that, but it's kind of not the right approach here. Rather, if you're looking through the array of elements for say a StringElement, to manipulate all string elements, you would do so with a [_user-defined type predicate_](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) (we could easily create those along with each element type, but they're also trivial to define when needed):

```ts
function isTextElement(e: Element): e is TextElement {
  return e.type === 'text';
}
const textElements = page.elements.filter(isTextElement);
```

The latter variable here now has `TextElement[]` type. It can also be defined inline:

```
page.elements.filter((e: Element): e is TextElement => e.type === 'text');
```

That `x is Foo` syntax looks weird, but it works, so 🤷.

Reply:
What if it's a function that actually accepts all types, for example:
```
function getLayerName(element) {
  if (element.layerName) {
    return element.layerName;
  }

  if (element.isBackground) {
    return __('Background', 'web-stories');
  }

  return getDefinitionForType(element.type).getLayerText(element);
}
```

If we would assign just `element: Element`, it would complain about `isBackground` not existing (it exists on shapes and media only).

Thoughts?